done.js

```diff
@@ -3,241 +3,336 @@ var chroma = require('@v3rse/chroma');
+            if (err) {
+                displayError("Git commit not possible - did you add files and include a message?");
+                return;
+            }
+            // If nothing had been added with git add
+            if (stdout.startsWith("On branch")) {
+                displayError("Git commit not possible - did you add files and include a message?");
+            }
+
+            // extract sha1 hash
+            var shaRegex = /^\[(.*) (.*)\]/gm;
+            var arr = shaRegex.exec(stdout);
+            const commitSha = arr[2];
+            //Alert the git commit has been created
+            console.log("Committed to git with SHA " + commitSha);
+            data.uncompleted[task].commit = commitSha;
+
+            //save data
+            setData(data);
+            console.log("Saved commit to task " + task + 1);
+        });
+
+    } else {
+        displayError("No such task");
+    }
+}
 
+function processGit() {
+    switch (argument) {
+        // just done git
+        case undefined:
+            listAll();
+            break;
+        // Create commit
+        case "commit":
+            gitCommit();
+            break;
+        // Add db to gitignore
+        case "init":
+            gitInit();
+            break;
+    }
+}
 
 //Utils
 
 //Formating for errors
-function displayError(string){
-	console.log(chroma.bgred(chroma.black(string)));
+function displayError(string) {
+    console.log(chroma.bgred(chroma.black(string)));
 }
 
 //Prints pending tasks
-function printUncompleted(data){
-	if (data.uncompleted.length) {
-		//print the uncompleted list. using ANSI colors and formating
-		console.log(chroma.underline.bgred("Pending:"));
-		data.uncompleted.forEach(function (task, index) {
-			console.log("\t",chroma.lyellow(index + 1 + ". ["),chroma.lred("✖"),chroma.lyellow("] "),chroma.italics.lblue(" ( Added " + moment(task.dateCreated).fromNow() + " ) "),task.task);
-		});
-	}
+function printUncompleted(data) {
+    if (data.uncompleted.length) {
+        //print the uncompleted list. using ANSI colors and formatting
+        console.log(chroma.underline.bgred("Pending:"));
+        data.uncompleted.forEach(function (task, index) {
+            const commit = task.commit || "";
+            console.log("\t", chroma.lyellow(index + 1 + ". ["), chroma.lred("✖"), chroma.lyellow("] "), chroma.italics.lblue(" ( Added " + moment(task.dateCreated).fromNow() + " ) "), task.task, chroma.lmagenta(commit));
+        });
+    }
 }
 
-
+//checks if the file exists.
+//If it does, it just calls back.
+//If it doesn't, then the file is created.
+// credit to https://stackoverflow.com/questions/12899061/creating-a-file-only-if-it-doesnt-exist-in-node-js
+function checkForFile(fileName, callback) {
+    fs.exists(fileName, function (exists) {
+        if (exists) {
+            callback();
+        } else {
+            fs.writeFile(fileName, {flag: 'wx'}, function (err, data) {
```

Comment:
I believe the options i.e. `{flag: 'wx'}` are supposed to be the third argument. The second argument is supposed to be the data written to the file. Check out the docs [here](http://devdocs.io/node/fs#fs_fs_writefile_file_data_options_callback). This currently writes `[object Object]` to the file.